server/remote/client.go

```diff
@@ -3,11 +3,17 @@
 
 package remote
 
-import "time"
+import (
+	"io"
+	"net/url"
+	"time"
+)
 
 type Client interface {
 	AcceptUserEvent(userID, eventID string) error
-	Call(method, path string, in, out interface{}) (responseData []byte, err error)
+	Call(method, path, contentType string, in io.Reader, out interface{}) (responseData []byte, err error)
```

Comment:
see my comment below, maybe remove `Call` as redundant? (This would also ensure that we always set the right content type in the calls.